x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -627,6 +644,22 @@ public boolean isAllowedByLicense(OperationMode minimumMode, boolean needActive)
         });
     }
 
+    /**
+     * Test whether current license expires in less than {@code GRACE_PERIOD_DURATION}.
```

Comment:
Why is this dependent on the `GRACE_PERIOD` ? It seems to be a co-incidence that they're both 7days, but they shouldn't be tied to one another.

Reply:
Good point, I'll add new const